components/cdc/src/endpoint.rs

```diff
@@ -476,7 +470,8 @@ impl<T: 'static + RaftStoreRouter<E>, E: KvEngine> Endpoint<T, E> {
                     conn.take_downstreams().into_iter().for_each(
                         |(region_id, (downstream_id, _))| {
                             if let Some(delegate) = self.capture_regions.get_mut(&region_id) {
-                                if delegate.unsubscribe(downstream_id, None) {
+                                delegate.unsubscribe(downstream_id, None);
```

Comment:
Looks like this PR fixed #10091 where when a TCP is disconnected, txn layer still adds old value to cache. 

Do we have tests to cover this case?

Reply:
Done.